modules/mqtt/destination/mqtt-destination.c

```diff
@@ -298,30 +172,43 @@ _init(LogPipe *d)
       return FALSE;
     }
 
-#if !SYSLOG_NG_HAVE_PAHO_HTTP_PROXY
-  if (self->http_proxy)
-    {
-      msg_warning_once("WARNING: the http-proxy() option of the mqtt() destination "
-                       "is not supported on the current libpaho-mqtt version. "
-                       "If you would like to use this feature, update to at least libpaho-mqtt 1.3.7");
-      g_free(self->http_proxy);
-      self->http_proxy = NULL;
-    }
-#endif
+  if (!mqtt_client_options_checker(&self->options))
+    return FALSE;
 
   if (!log_threaded_dest_driver_init_method(d))
     {
       return FALSE;
     }
 
-  if (_topic_name_is_a_template(self) && self->fallback_topic_name == NULL)
+  if (_topic_name_is_a_template(self) && self->fallback_topic == NULL)
     {
       msg_error("mqtt: the fallback_topic() argument is required if topic is templated for mqtt destinations",
                 evt_tag_str("driver", self->super.super.super.id),
                 log_pipe_location_tag(&self->super.super.super.super));
       return FALSE;
     }
 
+  if (mqtt_client_options_get_client_id(&self->options) == NULL)
+    {
+      gchar *tmp_client_id;
+
+      GlobalConfig *cfg = log_pipe_get_config(d);
+
+      if (cfg_is_config_version_older(cfg, VERSION_VALUE_3_34))
+        {
+          msg_warning_once("MQTT WARNING: you are using the old version, the default client id is different in the newer config",
+                           evt_tag_str("clint_id", _format_persist_name(d)),
+                           evt_tag_str("driver", self->super.super.super.id),
+                           log_pipe_location_tag(&self->super.super.super.super));
+          tmp_client_id = g_strdup(_format_persist_name(d));
+        }
+      else
+        tmp_client_id = g_strdup_printf("syslog-ng-%s-%s", "destination", self->topic_name->template);
+
+      mqtt_client_options_set_client_id(&self->options, tmp_client_id);
+      g_free(tmp_client_id);
```

Comment:
Can you move these to a separate function? Thanks!
Also, you can replace the first `%s` with `destination` directly.

Reply:
commit: dfb004c0914532621719c86d4051fbdd804bcd67